### PR TITLE
Wrap Overlay and Lightbox plugins popup in Dialog for screen reader c…

### PR DESCRIPTION
**Pull request by Kevin-WZW-Wu**
## What does this pull request (PR) do? / Que fait cette demande « pull » (PR)?
Wrap Overlay and Lightbox plugins popup in Dialog for screen reader compatibility

## Additional information (optional) / Information additionnelle (optionel)
I recently assessed an application using the WET4 light boxes (Lightbox) for the Session Timeout and some informational messages. The developer says they are using WET4 correctly. 
The issue is that I can navigate to the background of the Lightbox using the arrow keys on screen readers (JAWS 2020 and NVDA) in their application. 
I tested the Lightbox and Session Timeout on WET4, and I correctly am not able to navigate to the background while using a screen reader.
In my research, I discovered that adding `<div role="dialog">` around the `<div role="document">` will prevent the screen reader user from navigating to the background using the arrow keys.

**General checklist / Liste de contrôle générale**
Make your own list for the purpose of your Pull request. /// Faites votre propre liste en fonction des besoins de votre demande « pull ».

- [x] Build and test PR's code /// Rouler le script de compilation et tester le code de la PR
- [x] Validate changes against WCAG for accessibility /// Valider les changements avec WCAG pour l'accessibilité

**Related issues / Requêtes associées**

**Screenshots / Captures d'écrans**

### CODE REVIEW

File: src/plugins/lightbox/lightbox.js
Comment (GormFrank) [inline src/plugins/lightbox/lightbox.js:147]:
```suggestion
						$container = $wrap.find( ".mfp-container" ),
```

File: src/plugins/lightbox/lightbox.js
Comment (GormFrank) [inline src/plugins/lightbox/lightbox.js:172]:
```suggestion
					if ( $container.html().trim() !== "" ) {
```

**Review by GormFrank — CHANGES_REQUESTED** [review]
Before we do a more in-depth review of the code and test locally, the build presents five linting errors that can quickly be fixed. They are located in the `src/plugins/lightbox/lightbox.js` file:
pr-9248-build-errors

File: src/plugins/lightbox/lightbox.js
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/lightbox.js]: thx. Updated

File: src/plugins/lightbox/lightbox.js
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/lightbox.js]: thx. Updated

File: src/plugins/lightbox/_base.scss
Comment (duboisp) [inline src/plugins/lightbox/_base.scss:134]: Remove the spaces before the curly brace

File: src/plugins/lightbox/lightbox.js
Comment (duboisp) [inline src/plugins/lightbox/lightbox.js:174]: Add a value for the open attribute, like `open="open"`

File: src/plugins/overlay/overlay.js
Comment (duboisp) [inline src/plugins/overlay/overlay.js:118]: Can you try to wrap it with the dialog element?

**Review by duboisp — CHANGES_REQUESTED** [review]

File: src/plugins/lightbox/lightbox.js
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/lightbox.js]: thx. updated

File: src/plugins/lightbox/_base.scss
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/_base.scss]: updated

File: src/plugins/overlay/overlay.js
Comment (Kevin-WZW-Wu) [reply src/plugins/overlay/overlay.js:118]: it is much easier to manipulate the role attribute here. You will have multiple dialogs in the page. unless you do wrap and unwrap all the time.

File: src/plugins/lightbox/lightbox.js
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/lightbox.js]: thx , updated

File: src/plugins/overlay/overlay.js
Comment (duboisp) [reply src/plugins/overlay/overlay.js:118]: Ideally we will use dialog, but let do that change later. I will create a new issue to get that fixed eventually.

**Review by duboisp — APPROVED** [review]
Approved after a successful local testing.

**Review by duboisp — CHANGES_REQUESTED** [review]
I tested locally with Firefox and 
* there is an overflow of `<dialog >` elements that happen when opening and closing modal dialog created with Lighbox.
* The modal dialog is not centered anymore, some CSS adjustment might be required
* The example "Full with inline content" is broken, it only show vertical block 
* The image galleries examples is broken too, probably it requires some CSS adjustment

Here the test working example compiled with the code of this PR: https://universallabs.org/wet/labs/pr9248/unmin/demos/lightbox/lightbox-en.html

(Zip file: universallabs.org/wet/labs/pr9248.zip )

**Comment by Kevin-WZW-Wu** [comment]
@duboisp , thanks for your comments. I made some CSS changes to adjust that. There was no such issue before. Anyway, please take a look at the changes I pushed

File: src/plugins/lightbox/_base.scss
Comment (duboisp) [inline src/plugins/lightbox/_base.scss:131]: Is those css are the same as the CSS class `.mfp-container`? Can we just add the CSS class `mfp-container` on the dialog element.

**Review by duboisp — CHANGES_REQUESTED** [review]

**Comment by duboisp** [comment]
I updated the working example with the latest code: https://universallabs.org/wet/labs/pr9248/unmin/demos/lightbox/lightbox-en.html

File: src/plugins/lightbox/_base.scss
Comment (Kevin-WZW-Wu) [reply src/plugins/lightbox/_base.scss:131]: no, it is not the same. We can't just add class mfp-container

**Review by duboisp — CHANGES_REQUESTED** [review]
See my suggested changes, it remove the need of duplicating all the mfp-container CSS.

After those small change, we will merge your PR.

Thank you

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
I updated the working example with the latest code
https://universallabs.org/wet/labs/pr9248/unmin/demos/lightbox/lightbox-en.html

**Review by GormFrank — APPROVED** [review]

**Pull request merged**
